plugins/proxy-backend/src/service/router.ts

```diff
@@ -58,7 +63,16 @@ function buildMiddleware(
   // Attach the logger to the proxy config
   fullConfig.logProvider = () => logger;
 
-  return createProxyMiddleware(fullConfig);
+  // Only permit the allowed HTTP methods if configured
+  const filter = (_pathname: string, req: http.IncomingMessage): boolean => {
+    if (fullConfig.allowedMethods !== undefined) {
+      return !!fullConfig.allowedMethods.find(m => m === req.method);
+    }
+
+    return true;
```

Comment:
`return fullConfig.allowedMethods?.includes(req.method) ?? true` perhaps :P

Reply:
Nice idea. But that would return `string | true` instead of `boolean`. Isn't there a contains function in typescript?

Reply:
Yuuuuup, it's called `include`, see @freben's example xP

Reply:
Damn. Overlooked that xD

Reply:
`req.method` is `string | undefined` but according to the docs I opted for a `!`